src/main/java/ch/saunah/saunahbackend/service/UserService.java

```diff
@@ -203,4 +205,63 @@ public ResponseEntity<JwtResponse> signIn(SignInBody signInBody) throws Exceptio
 
         return ResponseEntity.ok(new JwtResponse(jwtToken));
     }
+
+    /**
+     * Get the user according to the provided ID
+     *
+     * @param id the id of the user to find
+     * @return The required user
+     * @throws NotFoundException if no such user exists
+     */
+    public User getUser(int id) throws NotFoundException {
+        User user = userRepository.findById(id).orElse(null);
+        if (user == null){
+            throw new NotFoundException(String.format("User with id %d not found!", id));
+        }
+        return user;
+    }
+
+    /**
+     * Return a list of all Users
+     *
+     * @return list of users
+     */
+    public List<User> getAllUser() {
+        return (List<User>) userRepository.findAll();
+    }
+
+    /**
+     * Edit an already existing User
+     * @param id the id of the user to be edited
+     * @param signUpBody the parameters to be changed
+     * @return
+     */
+    public User editUser(int id, SignUpBody signUpBody) {
```

Comment:
Maybe we could refactor the class `SignUpBody` to something like `UserBody`, as it's now no longer just used for singup, but for general user management.

Reply:
Done